sigil.lic

```diff
@@ -0,0 +1,153 @@
+custom_require.call(%w[common common-arcana common-travel drinfomon equipmanager events])
+
+class Sigil
+  include DRC
+  include DRCA
+  include DRCT
+
+  def initialize
+    settings = get_settings
+    @target = variable[1]
+    @target_precision = settings.sigil_min_precision_threshold
+    @sigil_book = settings.sigil_book
+
+    @primary_sigils = ['abolition','congruence','induction','permutation','rarefaction']
+    @secondary_sigils = ['antipode','ascension','clarification','decay','evolution','integration','metamorphosis','nurture','paradox','unity']
+    
+    Flags.add('precision', /Precision:(\d+)/)
+    Flags.add('clarity', /Clarity:(\d+)/)
+    Flags.add('danger', /Danger:[ -]+(\[*]{0,20})/)
+    Flags.add('sanity', /(?=Sanity:([ ]*)(\*{0,20}))/)
+    Flags.add('resolve', /(?=Resolve:([ ]*)(\*{0,20}))/)
+    Flags.add('focus', /(?=Focus:([ ]*)(\*{0,20}))/)
+    Flags.add('moresigils', "Remnants of the sigil pattern linger,")
+
+    @equipment_manager = EquipmentManager.new
+    @equipment_manager.empty_hands
+    
+    @current_precision = 0
+
+    harvest_sigil
+  end
+
+  
+  def find_sigil?
+    case line = bput('perceive sigil', @primary_sigils, @secondary_sigils, 'Roundtime', 'Having recently been searched')
+    when 'Having recently been searched'
+      exit
+    when *@primary_sigils
+      return true if line.include?(@target)
+      find_sigil?
+    when *@secondary_sigils
+      return true if line.include?(@target)
+      return false
+    when 'Roundtime'
+      find_sigil?
+    end
+  end
+
+  def scribe_sigil
+    Flags.reset('moresigils')
+    bput('get my burin', 'You get', 'You are already')
+    bput('get my blank scrolls', 'You get', 'You are already','You pick up')
+    bput('scribe sigil', 'You need', 'You carefully scribe')
+    waitrt?
+    bput('stow burin', 'You put')
+    bput("get #{@sigil_book}", 'You get')
+
+    case bput("put sigil in my #{@sigil_book}", 'You add', 'You rearrange', 'completely full')
+    when 'completely full'
+      bput('stow sigil', 'You put')
+    end
+    bput('stow blank scrolls', 'You put')
+    bput("stow my #{@sigil_book}", 'You put')
```

Comment:
Replace lines 51 - 63 with:
```ruby
        DRCC.get_crafting_item('burin', @bag, @bag_items, @belt)
        DRCC.get_crafting_item('blank scrolls', @bag, @bag_items, @belt)
        bput('scribe sigil', 'You need', 'You carefully scribe')
        waitrt?
        DRCC.stow_crafting_item('burin', @bag, @belt)
        DRCC.get_crafting_item(@sigil_book, @bag, @bag_items, @belt)

        case bput("put sigil in my #{@sigil_book}", 'You add', 'You rearrange', 'completely full')
        when 'completely full'
          bput('stow sigil', 'You put')
        end
        DRC.bput('get blank scrolls', 'You pick up', 'What were')
        DRCC.stow_crafting_item('blank scrolls', @bag, @belt)
        DRCC.stow_crafting_item(@sigil_book, @bag, @belt)
```

Reply:
Updated lines 52- 66 (the whole method):
```ruby
  def scribe_sigil
    Flags.reset('moresigils')
    DRCC.get_crafting_item('burin', @bag, @bag_items, @belt)
    DRCC.get_crafting_item('blank scrolls', @bag, @bag_items, @belt)
    bput('scribe sigil', 'You need', 'You carefully scribe')
    waitrt?
    DRCC.stow_crafting_item('burin', @bag, @belt)
    DRCC.stow_sigil(@sigil_book, @bag, @bag_items, @belt)
    DRC.bput('get blank scrolls', 'You pick up', 'What were')
    DRCC.stow_crafting_item('blank scrolls', @bag, @belt)
    @equipment_manager.empty_hands
    if Flags['moresigils']
      scribe_sigil
    end
  end
```

`common-crafting` addition:  lines 245 - 255:
```ruby
  def stow_sigil(tome, bag, bag_items, belt)
    $ORDINALS.each do |ordinal|
      get_crafting_item("#{ordinal} #{tome}", bag, bag_items, belt)
      unless DRCI.put_away_item?('sigil', tome)
        stow_crafting_item(tome, bag, belt)
        next
      end
      stow_crafting_item(tome, bag, belt)
      break
    end
  end
  ```
  May change it so it looks for sigil book|tome|any other name for it. Not Sure yet. Or, make `sigil_book` an array and loop through that.